modules/navier_stokes/src/postprocessors/InternalVolumetricFlowRate.C

```diff
@@ -0,0 +1,162 @@
+                             "'upwind' and 'average', with the default being 'upwind'.");
+  params.addCoupledVar(
+      "advected_variable", 0, "The advected variable quantity of which to study the flow");
+  params.addParam<MaterialPropertyName>(
+      "advected_mat_prop", 0, "The advected material property of which to study the flow");
+  return params;
+}
+
+InternalVolumetricFlowRate::InternalVolumetricFlowRate(const InputParameters & parameters)
+  : InterfaceIntegralPostprocessor(parameters),
+    _fv(getParam<bool>("fv")),
+    _vel_x(coupledValue("vel_x")),
+    _vel_y(coupledValue("vel_y")),
+    _vel_z(coupledValue("vel_z")),
+    _fv_vel_x(dynamic_cast<const MooseVariableFV<Real> *>(getFieldVar("vel_x", 0))),
+    _fv_vel_y(dynamic_cast<const MooseVariableFV<Real> *>(getFieldVar("vel_y", 0))),
+    _fv_vel_z(dynamic_cast<const MooseVariableFV<Real> *>(getFieldVar("vel_z", 0))),
+    _advected_variable(coupledValue("advected_variable")),
+    _fv_advected_variable(
+        dynamic_cast<const MooseVariableFV<Real> *>(getFieldVar("advected_variable", 0))),
+    _advected_material_property(getADMaterialProperty<Real>("advected_mat_prop")),
+    _advected_material_property_neighbor(getNeighborADMaterialProperty<Real>("advected_mat_prop"))
+{
+  /// Check that at most one advected quantity has been provided
+  if (parameters.isParamSetByUser("advected_variable") &&
+      parameters.isParamSetByUser("advected_mat_prop"))
+    mooseError(
+        "InternalVolumetricFlowRatePostprocessor should be provided either an advected variable or "
+        "an advected material property");
+
+  using namespace Moose::FV;
+
+  const auto & advected_interp_method = getParam<MooseEnum>("advected_interp_method");
+  if (advected_interp_method == "average")
+    _advected_interp_method = InterpMethod::Average;
+  else if (advected_interp_method == "upwind")
+    _advected_interp_method = InterpMethod::Upwind;
+  else
+    mooseError("Unrecognized advected quantity interpolation type ",
+               static_cast<std::string>(advected_interp_method));
+}
+
+Real
+InternalVolumetricFlowRate::computeQpIntegral()
+{
+#ifdef MOOSE_GLOBAL_AD_INDEXING
+  if (_fv)
+  {
+    /// We should not be at a boundary
+    const FaceInfo * const fi = _mesh.faceInfo(_current_elem, _current_side);
+    mooseAssert(fi, "We should have a face info");
+    mooseAssert(
+        !fi->isBoundary(),
+        "Use VolumetricFlowRate instead of InternalVolumetricFlowRate for domain boundaries");
+
+    const Elem * neighbor = _current_elem->neighbor_ptr(_current_side);
+
+    /// Get face value for velocity
+    // FIXME Make sure getInternalFaceValue uses the right interpolation method, see #16585
+    const auto & vx_face =
+        !getFieldVar("vel_x", 0)
+            ? _vel_x[_qp]
+            : MetaPhysicL::raw_value(_fv_vel_x->getInternalFaceValue(neighbor, *fi, _vel_x[_qp]));
+
+    const auto & vy_face =
+        !getFieldVar("vel_y", 0)
+            ? _vel_y[_qp]
```

Comment:
```suggestion
            ? Real(0)
```

Reply:
This one is not required, so the ternary is ok I think, but I would make it even more transparent about what is happening, e.g. the explicit assignment to `Real(0)`. (I think that will compile fine, but not 100% sure)

Reply:
see above